doc: Add documentation for SSH port forward.

scottilee opened the pull request:
<!-- What's this PR for?  (Just a link to an issue is fine.) -->
https://github.com/zulip/zulip/issues/10696

**Testing Plan:** <!-- How have you tested? -->
Made the changes on a development server and ran `./tools/build-docs`. Screenshot of change below.

**GIFs or Screenshots:** <!-- If a UI change.  See:
  https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
  -->
<img width="717" alt="screen shot 2018-10-19 at 9 15 55 pm" src="https://user-images.githubusercontent.com/7443383/47251179-367e0480-d3e5-11e8-8e31-8ed0825b9052.png">

<!-- Also be sure to make clear, coherent commits:
  https://zulip.readthedocs.io/en/latest/contributing/version-control.html
  -->

timabbott commented:
I think we should put those docs inside `docs/development/remote.md` and then link to them from here, because the `remote.md` page has other advice that can be helpful for remote development.

scottilee commented:
@timabbott I moved the changes to the remote.md section and added a link in the setup-advanced.md section.

<img width="718" alt="screen shot 2018-10-22 at 1 08 32 pm" src="https://user-images.githubusercontent.com/7443383/47316461-0bb9c900-d5fc-11e8-950d-cf84a607eee3.png">

timabbott commented:
Nice, merged as b4f95aa12420f11afd383746af3ecee0803e4ee3 after a bit of editing, thanks @scottilee!
